DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -755,9 +778,15 @@ private void HandleOptionsRequest(string playerName, int options)
             if (color < 0 || color > MPColors.Count)
                 return;
 
-            var disallowedSides = GetDisallowedSides();
+            var disallowedSides = GetDisallowedSides().ToList();
+
+            // Disallowed sides from client, maps, or game modes do not take random selectors into account
+            // So, we need to insert "false" for each random at the beginning of this list AFTER getting them
+            // from client, maps, or game modes.
+            for(int i = 0; i < RandomSelectorCount; i++)
+                disallowedSides.Insert(0, false);
 
-            if (side > 0 && side <= SideCount && disallowedSides[side - 1])
+            if (side > 0 && side <= SideCount && disallowedSides[side])
```

Comment:
Eh, what? `side` (array id element) cannot be equal `SideCount` because it would be crash due to bounds error!

![image](https://github.com/user-attachments/assets/2cee9dbb-b515-429a-8017-c94ece882159)

Reply:
f0da740

still, we need to go through the original issue it tries to resolve